examples/infix_math_parser.py

```diff
@@ -0,0 +1,190 @@
+
+    # Supported infix binary operators, i.e., '1+1'. The key is the notation of the operator in infix format,
+    # and the value the notation in parsed format.
+    BINARY_OPERATORS: ClassVar[dict] = {
+        "+": "Add",
+        "-": "Subtract",
+        "*": "Multiply",
+        "/": "Divide",
+        "**": "Power",
+    }
+
+    # Supported infix unary operators, i.e., 'Cos(90)'. The key is the notation of the operator in infix format,
+    # and the value the notation in parsed format.
+    UNARY_OPERATORS: ClassVar[dict] = {
+        "Cos": "Cos",
+        "Sin": "Sin",
+        "Tan": "Tan",
+        "Exp": "Exp",
+        "Ln": "Ln",
+        "Lb": "Lb",
+        "Lg": "Lg",
+        "LogOnePlus": "LogOnePlus",
+        "Sqrt": "Sqrt",
+        "Square": "Square",
+        "Abs": "Abs",
+        "Ceil": "Ceil",
+        "Floor": "Floor",
+        "Arccos": "Arccos",
+        "Arccosh": "Arccosh",
+        "Arcsin": "Arcsin",
+        "Arcsinh": "Arcsinh",
+        "Arctan": "Arctan",
+        "Arctanh": "Arctanh",
+        "Cosh": "Cosh",
+        "Sinh": "Sinh",
+        "Tanh": "Tanh",
+        "Rational": "Rational",
+        "-": "Negate",
+    }
+
+    # Supported infix variadic operators (operators that take one or more comma separated arguments),
+    # i.e., 'Max(1,2, Cos(3)). The key is the notation of the operator in infix format,
+    # and the value the notation in parsed format.
+    VARIADIC_OPERATORS: ClassVar[dict] = {"Max": "Max"}
+
+    def __init__(self):
+        """A parser for infix notation, e.g., the human readable way of notating mathematical expressions.
+
+        The parser can parse infix notation stored in a string. For instance,
+        "Cos(2 + f_1) - 7.2 + Max(f_2, -f_3)" is parsed to the list:
+        ['Cos', [[2, '+', 'f_1']]], '-', 7.2, '+', ['Max', ['f_2', ['-', 'f_3']].
+
+        """
+        # Scope limiters
+        lparen = Suppress("(")
+        rparen = Suppress(")")
+
+        # Define keywords (Note that binary operators must be defined manually)
+        symbols_variadic = set(InfixExpressionParser.VARIADIC_OPERATORS.keys())
+        symbols_unary = set(InfixExpressionParser.UNARY_OPERATORS.keys())
+
+        # Define binary operation symbols (this is the manual part)
+        # If new binary operators are to be added, they must be defined here.
+        signop = one_of("+ -")
+        multop = one_of("* /")
+        plusop = one_of("+ -")
+        expop = Literal("**")
+
+        # Dynamically create Keyword objects for variadric functions
+        variadic_pattern = r"\b(" + f"{'|'.join([*symbols_variadic])}" + r")\b"
+        variadic_func_names = Regex(variadic_pattern)
+
+        # Dynamically create Keyword objects for unary functions
+        unary_pattern = r"\b(" + f"{'|'.join([*symbols_unary])}" + r")\b"
+        unary_func_names = Regex(unary_pattern)
+
+        # Define operands
+        integer = pyparsing_common.integer.set_parse_action(
```

Comment:
Adding parse actions to integer and scientific is not necessary, since the expressions in pyparsing.common already have these same parse actions.

Reply:
This is nice to know, I guess this also improves performance a little bit to not add explicit parse actions? Anyhow, I removed them from the example.